### PR TITLE
feat: comprehensive infrastructure improvements

### PR DESCRIPTION
**Pull request by docdyhr**
This PR implements comprehensive infrastructure improvements including enhanced branch protection rules, security monitoring with Dependabot, workflow consistency fixes, and comprehensive documentation.

## Summary by Sourcery

Implement comprehensive infrastructure and build system improvements, enhance CI workflows with Python 3.13 and security auditing, enforce consistent branch protection and security monitoring, and add supporting scripts, documentation, and extensive tests.

New Features:
- Add Python 3.13 support including CI matrix update and compatibility docs
- Introduce daily security audit workflow and CodeQL analysis for proactive vulnerability scanning
- Add advanced auto-update test suite covering edge cases and integration scenarios

Enhancements:
- Unify branch references to use 'master' across all GitHub Actions workflows and add branch protection and verification scripts
- Revise Dependabot configuration for daily security updates with grouping and auto-merge of patch/minor fixes
- Consolidate multi-line print and function signature formatting for code consistency

Build:
- Update pyproject.toml classifiers for Python 3.13
- Add requirements-py313.txt and lock file for deterministic dependency installs

CI:
- Update CI matrix to include Python 3.13 with experimental support
- Add security-audit.yml and codeql-analysis.yml workflows for automated security scanning
- Refine lint, security, performance workflows to trigger only on the 'master' branch

Documentation:
- Add extensive infrastructure documentation: branch protection, security monitoring, Python 3.13 support, and release guides
- Create SECURITY.md vulnerability disclosure policy and CODEOWNERS for review management
- Provide example and support scripts under docs and examples for Python 3.13 and setup tasks

Tests:
- Expand test coverage with platform compatibility, auto-update integration, and edge-case tests
- Add tests for security workflows and CI configuration consistency
- Include serialization of print and formatting changes in existing test suites

Chores:
- Introduce multiple setup and verification shell scripts for branch protection, workflow cleanup, and security setup

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Install dependencies
+        uses: ./.github/actions/setup-python-deps
+
+      - name: Autobuild (no-op for Python)
+        uses: github/codeql-action/autobuild@v3
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v3
+        with:
+          category: 'security-and-quality'
```
[the remark this answers is not in the document]
Comment (docdyhr) [reply .github/workflows/codeql-analysis.yml:40]: @claude

File: setup-security-monitoring.sh [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+echo "🚨 Enabling Dependabot alerts..."
+gh api \
+    --method PUT \
+    -H "Accept: application/vnd.github+json" \
+    -H "X-GitHub-Api-Version: 2022-11-28" \
+    "/repos/$REPO/vulnerability-alerts" || echo "⚠️  Dependabot alerts may already be enabled"
+
+# Check if GitHub Advanced Security is available (may require GitHub Pro/Enterprise)
+echo "🔍 Checking GitHub Advanced Security availability..."
+ADVANCED_SECURITY=$(gh api "/repos/$REPO" --jq '.security_and_analysis.advanced_security.status' 2>/dev/null || echo "not_available")
```
[the remark this answers is not in the document]
Comment (docdyhr) [reply setup-security-monitoring.sh:47]: @claude

File: setup-security-monitoring.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+gh api \
+    --method PATCH \
+    -H "Accept: application/vnd.github+json" \
+    -H "X-GitHub-Api-Version: 2022-11-28" \
+    "/repos/$REPO" \
+    -f "has_vulnerability_alerts=true" || echo "⚠️  Vulnerability alerts may already be enabled"
+
+# Enable Dependabot security updates
+echo "🤖 Enabling Dependabot security updates..."
+gh api \
```
[the remark this answers is not in the document]
Comment (docdyhr) [reply setup-security-monitoring.sh:31]: @claude

**Comment by docdyhr** [comment]
✅ **Status Update**: All Sourcery AI feedback has been addressed:

- ✅ Python 3.12+ syntax compatibility fixed with proper conditional imports
- ✅ Shell script portability issues were already resolved
- ✅ PyYAML availability checks already implemented
- ✅ GitHub API error handling already improved
- ✅ All CI checks are now passing

The PR is ready for merge!

**Comment by docdyhr** [comment]
@sourcery-ai resolve
